pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -3026,7 +3026,7 @@ public void onUpdate(TopicPolicies policies) {
             updateSubscribeRateLimiter();
             replicators.forEach((name, replicator) -> replicator.getRateLimiter()
                     .ifPresent(DispatchRateLimiter::updateDispatchRate));
-
+            checkMessageExpiry();
```

Comment:
Is it able to add a test to avoid regression?

Reply:
Yes. I have added UT `testTTLPoliciesUpdate`. PTAL again @codelipenghui